app/javascript/components/upload/WizardNavPanel.js

```diff
@@ -50,5 +57,75 @@ function RawWizardNavPanel({
   </div>
 }
 
+/** shows current expression and clustering visualization status */
+function VisualizationStatuses({ serverState }) {
+  if (!serverState) {
+    return <></>
+  }
+  const metadataFiles = serverState.files.filter(metadataFileFilter)
+  const isMetadataParsing = metadataFiles.some(f => f.parse_status === 'parsing')
+  const isMetadataParsed = metadataFiles.some(f => f.parse_status === 'parsed')
+
+  const clusteringFiles = serverState.files.filter(clusterFileFilter)
+  const isClusteringParsing = isMetadataParsing || clusteringFiles.some(f => f.parse_status === 'parsing')
+  const isClusteringParsed = isMetadataParsed && clusteringFiles.some(f => f.parse_status === 'parsed')
+
+  const processedExpFiles = serverState.files.filter(processedFileFilter)
+  const isExpressionParsing = isMetadataParsing || processedExpFiles.some(f => f.parse_status === 'parsing')
+  const isExpressionParsed = isMetadataParsed && processedExpFiles.some(f => f.parse_status === 'parsed')
+
+  let clusterStatusMsg = <span className="success">clustering visuals <FontAwesomeIcon icon={faCheck}/></span>
+  if (!isClusteringParsed) {
+    clusterStatusMsg = <span className="warning">no cluster visuals <FontAwesomeIcon icon={faQuestionCircle}/></span>
+  }
+  const clusterVizIndicator = <div>
+    <OverlayTrigger
+      trigger={['hover', 'focus']}
+      rootClose placement="top"
+      overlay={clusteringHelpContent(isClusteringParsing)}>
+      <span>
+        {clusterStatusMsg} { isClusteringParsing && <LoadingSpinner/> }
+      </span>
+    </OverlayTrigger>
+  </div>
+
+  let expressionStatusMsg = <span className="success">expression visuals <FontAwesomeIcon icon={faCheck}/></span>
+  if (!isExpressionParsed) {
+    expressionStatusMsg = <span className="warning">no expression visuals <FontAwesomeIcon icon={faQuestionCircle}/></span>
+  }
+  const expressionVizIndicator = <div>
+    <OverlayTrigger
+      trigger={['hover', 'focus']}
+      rootClose placement="top"
+      overlay={expressionHelpContent(isExpressionParsing)}>
+      <span>
+        {expressionStatusMsg} { isExpressionParsing && <LoadingSpinner/> }
+      </span>
+    </OverlayTrigger>
+  </div>
+  return <div className="viz-notifications">
+    {clusterVizIndicator}
+    {expressionVizIndicator}
+  </div>
+}
+
+/** gets the popup message based on whether there are files parsing */
+function expressionHelpContent(isExpressionParsing) {
```

Comment:
Technically, to view expression-based visualizations, you need all 3 main file types: expression, metadata, and clustering.  All of our non-genomic visualizations require a cluster to view.